crates/ruff_python_ast/src/str.rs

```diff
@@ -67,15 +67,55 @@ pub fn is_triple_quote(content: &str) -> bool {
     TRIPLE_QUOTE_STR_PREFIXES.contains(&content) || TRIPLE_QUOTE_BYTE_PREFIXES.contains(&content)
 }
 
+/// Return `true` if the string expression is an implicit concatenation.
+pub fn is_implicit_concatenation(content: &str) -> bool {
+    let leading_quote_str = leading_quote(content).unwrap();
+    let trailing_quote_str = trailing_quote(content).unwrap();
+
+    // Short-circuit: if the trailing quote doesn't match the _expected_ trailing quote, then
+    // the string is implicitly concatenated.
+    if trailing_quote_str != trailing_quote(trailing_quote_str).unwrap() {
+        return true;
+    }
```

Comment:
D'oh, this is supposed to be `trailing_quote(leading_quote_str)` on the right-hand side. Wasn't caught by tests since it's just a short-circuit...

To be clear, the trick is that the leading quote includes the string "kind" prefix.

So given a string like:

```py
u"""abc def"""
```

The leading quote would be `u"""`, and the trailing quote would be `"""`.

Reply:
I'm still somewhat confused. Shouldn't it be `leading_quote_str != trailing_quote_str`?

Reply:
No, because `u"""` and `"""` is a valid leading-trailing quote pair.

What we want to know is that if the string has a leading quote of `u"""`, then its expected trailing quote is `"""`. Here, we're checking if that expected trailing quote matches the _actual_ trailing quote.

For example, given:

```
u"""abc""" 'def'
```

The leading quote would be `u"""`. The trailing quote would be `'`. The _expected_ trailing quote would be `"""`. Since `'` does not equal `"""`, we'd return `false`.

Reply:
I'd like to make this clearer since it's evidently too confusing :joy: Any suggestions?

Reply:
Uhh, python quotes are confusing. Why so many :confused: 

Paste your reply as a comment :)